libmamba/include/mamba/core/context.hpp

```diff
@@ -118,11 +118,58 @@ namespace mamba
     {
     public:
 
-        std::string caller_version = "";
-        std::string conda_version = "3.8.0";
-        std::string current_command = "mamba";
-        std::string custom_banner = "";
-        bool is_micromamba = false;
+        struct RemoteFetchInfo
+        {
+            // ssl_verify can be either an empty string (regular SSL verification),
+            // the string "<false>" to indicate no SSL verification, or a path to
+            // a directory with cert files, or a cert file.
+            std::string ssl_verify{ "" };
+            bool ssl_no_revoke{ false };
+            bool curl_initialized{ false };  // non configurable, used in fetch only
+
+            std::string user_agent{ "mamba/" LIBMAMBA_VERSION_STRING };
+
+            int connect_timeout_secs{ 10 };
+            // int read_timeout_secs { 60 };
+            int retry_timeout{ 2 };  // seconds
+            int retry_backoff{ 3 };  // retry_timeout * retry_backoff
+            int max_retries{ 3 };    // max number of retries
+        };
+
+        struct OutputInfo
+        {
+            int verbosity{ 0 };
+            log_level logging_level{ log_level::warn };
+
+            bool json{ false };
+            bool quiet{ false };
+
+            std::string log_pattern{ "%^%-9!l%-8n%$ %v" };
+            std::size_t log_backtrace{ 0 };
+        };
+
+        struct DesignInfo
```

Comment:
Did you meant "graphics" here? Design isn´t necessarilly about graphics.

Reply:
Or maybe "display"?

Reply:
Graphics or Display then? :)

Reply:
Both are fine to me, also this will change once we separate the logic from the "view" (which is another possible name XD)
I'll let you chose. My initial comment is because "design" can mean "concept" but the intention, if I understand correctly, is more like "the visible thing we will show", so any word more like "view", "graphcs" et.c works for me.